### DISCUSSION TITLE
pyright still reports PathLike as unknown

### DISCUSSION THREAD

**Discussion by dd-ssc** [gitpython-developers/GitPython#1622] [Q&A] [answered]
With sample code like

```python
from git.repo import Repo
repo = Repo(root_of_my_repo)
print(f'repo working dir: {repo.working_tree_dir}')
print(f'is repo dirty ?: {repo.is_dirty()}')
```

I see VS Code / pylance / pyright messages like these (shortened for better readability):

```
Type of "working_tree_dir" is partially unknown
  Type of "working_tree_dir" is "Unknown | None"
```

```
Type of "is_dirty" is partially unknown
  Type of "is_dirty" is "(index: bool = True, ..., path: Unknown | None = None) -> bool"
```

both of which seem to be caused by the type of `PathLike` to be unknown.

However, `PathLike` clearly _is_ properly defined at https://github.com/gitpython-developers/GitPython/blob/1c8310d7cae144f74a671cbe17e51f63a830adbf/git/types.py#L46.

Also, in general, I see from https://github.com/gitpython-developers/GitPython/issues/1095 that full type support has been added to GitPython two years ago - and some detail of this very `PathLike` issue seems to have been addressed in https://github.com/gitpython-developers/GitPython/issues/1473 about 13 months ago.

AFAICT, I use the latest versions for anything involved:

```
GitPython: 3.1.32
macOS:     13.4.1
Python:    3.10.12
Pylance:   v2023.7.40 (VS Code extension)
pyright:   1.1.318
```

Why am I still seeing these pyright issues ?
Do I need to do anything special to get GitPython type support ?
Could there anything wrong with my setup ?
(Using the same setup (e.g. pyright conf) as in 10+ similar projects with no problems, but you never know...)

Really puzzled here and would be really grateful for any advice.

Thank you very much.

**Comment by Byron** [comment, marked as answer]
Since your setup works fine for 10+ similar projects, I'd assume that those simply are better maintained. As a matter of fact, I have no development setup for Python at all and cannot run any tests locally anymore.
As for contributors, I don't know how they do it but would assume they have similar issues that they ignore.

After the typing extensions were contributed, they are essentially unmaintained, but receive some improvements from time to time from folks who experience issues with them.

Thus you are welcome to fix it for your setup and submit a PR - I have no idea how that should work though.
